core/controllers/base.py

```diff
@@ -238,9 +238,9 @@ def dispatch(self):
                         'Registration session expired.')
                 csrf_token = self.request.get('csrf_token')
                 if not csrf_token:
-                    raise Exception(
-                        'Missing CSRF token. Changes were not saved. '
-                        'Please report this bug.')
+                    raise self.UnauthorizedUserException(
```

Comment:
Please make sure that there is a backend test for this.

Every time you change a file, think about tests -- either existing tests need to be modified, or new ones need to be written.

Reply:
Hi @seanlip ,

This has passed the back-end tests locally. Also,  i dint find a necessity to write a back-end test the existing ones seems to work fine for this too.
I'll have a check on this again Thanks :)

Reply:
It seems like the behaviour has changed from 500 to 401. I would expect a backend test to catch that.

Reply:
Done